### PR TITLE
feat(config): Implement std::default::Default

### PR DESCRIPTION
**Pull request by BKDaugherty**
Hey! I'm hoping to use this library for a project of mine, and wanted to instantiate a Config using the Default trait. Thought I'd put up a PR for it! Lmk what you think!

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -33,27 +33,33 @@ pub struct Config {
     pub single_port: bool,
     /// Refuse all write requests, making the server read-only. (default: false)
     pub read_only: bool,
-    /// Duplicate all packets sent from the server. (default: 1)
+    /// Duplicate all packets sent from the server. (default: 0)
```
Comment (stappersg) [inline src/config.rs:36]: Do this a separate merge request. (You all ready made a separate commit.)

Comment (BKDaugherty) [reply src/config.rs]: https://github.com/altugbakan/rs-tftpd/pull/20

**Comment by stappersg** [comment]
(Here not the project lead, some random Joe)
I see only lines moved, I don't see any improvement.
Most likely due me not understanding the _"I'd like to use `Config::default` in my own code when constructing a `Config`. Let's implement the Default trait so that others can use it without wrapping our struct due to ownership rules."_ commit message.

**Comment by BKDaugherty** [comment]
> Most likely due me not understanding the "I'd like to use Config::default in my own code when constructing a Config. Let's implement the Default trait so that others can use it without wrapping our struct due to ownership rules." commit message.

Ah. @stappersg, no worries! Happy to explain :) 

It's fairly common in Rust for types to implement certain Traits, (or interfaces in other languages), for standard behaviors. One of those behaviors is constructing a default representation of a type. Check out the docs on the trait here: https://doc.rust-lang.org/std/default/trait.Default.html

I typically do this when I'd like to create a config with specific values, but I don't really care about the other values.

Let's take for example someone wanting to run their TFTP server on the IPV6 localhost (`::1`), but they want to just spin up the rest of the config to the default values. (Say for example, they don't know that the port should be `69`, or don't want to duplicate that constant in their own code.) They can use the following syntax

```rust
let config = Config {
  ip_address: IpAddr::V6(Ipv6Addr::LOCALHOST),
  ..Config::default()
};
```
And rely on the default implementation to do the rest.

It's possible to do something similar to the above with the existing `Config::new` method, but a little awkward, and unexpected compared to the rest of the ecosystem, which relies on the standard libraries trait.

```rust
    let config = Config {
        ip_address: std::net::IpAddr::V6(Ipv6Addr::LOCALHOST),
	..Config::new([].into_iter()).expect("Config building with no args should be fine, and my application depends on that")
    };
```

**Comment by altugbakan** [comment]
Looks great initially, I’ll take a closer look when I get home. Thank you for your addition!

File: src/config.rs [hunk 2]
```diff
@@ -39,21 +39,27 @@ pub struct Config {
     pub overwrite: bool,
 }
 
+impl std::default::Default for Config {
```
Comment (altugbakan) [inline src/config.rs:42]: Can we use `std::default::Default` on top and use only `Default` here?

**Review by altugbakan — COMMENTED** [review]
Looks great, I just only have one small styling request.

File: src/config.rs (hunk 2 above)
Comment (BKDaugherty) [reply src/config.rs:42]: Yeah for sure!

Comment (BKDaugherty) [reply src/config.rs:42]: Actually on second thought we don't even need to bring in the import. It's brought in as a part of `core` for edition 2021.

Comment (BKDaugherty) [reply src/config.rs:42]: (Which is why all of the other references to `Default` work even though I didn't scope them 🙃 )

Comment (altugbakan) [reply src/config.rs]: Oh nice haha

**Review by altugbakan — APPROVED** [review]
LGTM

**Pull request merged**